### PR TITLE
Help

### PR DESCRIPTION
**Pull request by k0789885**

### CODE REVIEW

File: k0789885/MyGame.py [hunk 1]
```diff
@@ -1,239 +1,202 @@
...
         print ""
-        print "Strength = 4"
-        print "Dexterity = 2"
-        print "Wisdom = 8"
+    elif weapon.selected_value == 2:
+        print ""
+    elif weapon.selected_value == 3:
+        print "" 
+    elif weapon.selected_value == 4:
+        thr1
```
Comment (chrisortman) [inline k0789885/MyGame.py:82]: I don't see this defined anywhere

File: k0789885/MyGame.py [hunk 2]
```diff
@@ -1,239 +1,202 @@
...
-while Class != "Warrior" and Class != "Archer" and Class != "Mage":
-    Class = raw_input("> ")
-    if Class == "1" or Class == "Warrior":           #Warrior
-        print "Grand, fan of hitting and kicking huh? Your stats are below."
-        print ""
-        print "Strength = 8"
-        print "Dexterity = 4"
-        print "Wisdom = 2"
+def weapon():
+    weapon = Menu()
```
Comment (chrisortman) [inline k0789885/MyGame.py:59]: While it's legal to do, I would be careful with using the same name to mean multiple things. It's easy to get confused about if you are talking about the `<type function>` weapon or the `<type Menu>`

**Review by chrisortman — COMMENTED** [review]
It seems to work for me as long as I don't choose Mace

As much as is possible try to keep your functions only dealing with things that are passed into them. You will have an easier time keeping track of what is going on then.

It feels like there's a missing piece...like when I look at the code and I see the way the menus work it makes me say if only there was this then it would be simpler. I think the missing thing might be called a 'Scene' or a 'Frame'

Basically a Scene/Frame might have some dialogue (Good morning lord maul ... ) and then a menu of choices. the choice seem to determine what the next scene should be, so maybe the scene should return next scene and then the game is looping through a list of scenes?

**Comment by k0789885** [comment]
Though that is true I want a direct story to the game but a few (mini
quest) or things you can do before you progress with the story. That's why
I have a part of code that says

If startgame.selected_value = 1
Print"..."
Elif
Print"..."

I wanted there to be a consequence for being late to training. And I want a
good amount of boolean to make the game a bit more interesting. Like you
get a different ending based on what you did in the game. I read the
chapter you mentioned in LPTHW and it went through a 1 line story with no
factors. That's why I didn't use it.

**Comment by chrisortman** [comment]
Understood. 

> I wanted there to be a consequence for being late to training. 

How can you encode this rule into the program....

Instead of 

`if startgame.selected_value = 1`

`if player.late_to_training`

Keep in mind that you want to be ale to break the problem down and solve it in pieces... just like we can start with

```
if True:
  print "Something"
```

And eventually turn it into

```
if result_of_some_function():
  print "Something"
```

Once you can get something working that let's you hard code the next scene you can replace the call that gives back the next scene with something that has a lot more _smarts_ by looking at attributes of player / world etc
